File: unitri-ppi/aulas-spring/src/main/java/com/example/aulasspring/service/PessoaService.java

package com.example.aulasspring.service;

import com.example.aulasspring.model.Pessoa;
import com.example.aulasspring.repository.PessoaRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;


@Service
public class PessoaService {

    @Autowired
    private PessoaRepository repository;

    public List<Pessoa> findAll() {
        return repository.findAll();
    }

    public Pessoa findOne(Long id) {
        return (Pessoa) repository.getOne(id);
    }

    public Pessoa save(Pessoa post) {
        return (Pessoa) repository.saveAndFlush(post);
    }

    public void delete(Long id) {
        repository.deleteById(id);
    }
}

File: unitri-ppi/aula-07/src/main/kotlin/com/example/aula07/service/PessoaService.kt
package com.example.aula07.service

import com.example.aula07.model.Pessoa
import com.example.aula07.repository.PessoaRepository
import org.springframework.stereotype.Service


@Service
class PessoaService(
    val repository: PessoaRepository
) {

    fun findAll(): List<Pessoa> = repository.findAll()

    fun findOne(id: Long): Pessoa = repository.getOne(id)

    fun save(post: Pessoa): Pessoa = repository.saveAndFlush(post)

    fun delete(id: Long) = repository.deleteById(id)
}